src/github.com/getlantern/flashlight/analytics/analytics.go

```diff
@@ -0,0 +1,68 @@
+package analytics
+
+import (
+	"net/http"
+
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/getlantern/analytics"
+	"github.com/getlantern/flashlight/ui"
+	"github.com/getlantern/golog"
+)
+
+const (
+	messageType = `Analytics`
+)
+
+var (
+	log        = golog.LoggerFor("flashlight.analytics")
+	service    *ui.Service
+	httpClient *http.Client
+)
+
+func Configure(autoReport *bool, newClient *http.Client) {
+	httpClient = newClient
+
+	if autoReport != nil && *autoReport {
+		err := Start()
+		if err != nil {
+			log.Errorf("Error starting analytics service: %q", err)
+		}
+	}
```

Comment:
I guess the salient question is whether or not the communication is always one way (i.e. client posting something to server).  If so, it's probably easiest just to make this a RESTful POST, and then you don't have to bother with starting/stopping services.  You just post from the client and, if reporting is enabled, the back-end posts that to the server.  If you want to get fancy, you can have the client eschew posting to the back-end when reporting is disabled, but I'd consider that icing on the cake.